include/deal.II/fe/fe_tools.templates.h

```diff
@@ -2916,6 +2916,79 @@ namespace FETools
 
 
 
+  namespace
+  {
+    // internal helper class to partially specialize the number type
+
+    template <int dim, int spacedim, typename number>
+    struct ConvertHelper
+    {
+      static void convert_generalized_support_point_values_to_dof_values(
+        const FiniteElement<dim, spacedim> &finite_element,
+        const std::vector<Vector<number> > &support_point_values,
+        std::vector<number>                &dof_values)
+      {
+        static Threads::ThreadLocalStorage<std::vector<Vector<double> > >
+        double_support_point_values;
+        static Threads::ThreadLocalStorage<std::vector<double> >
+        double_dof_values;
+
+        double_support_point_values.get().resize(support_point_values.size());
+        double_dof_values.get().resize(dof_values.size());
+
+        for (unsigned int i = 0; i < support_point_values.size(); ++i)
+          {
+            double_support_point_values.get()[i].reinit(
+              finite_element.n_components(), false);
+            std::copy(std::begin(support_point_values[i]),
+                      std::end(support_point_values[i]),
+                      std::begin(double_support_point_values.get()[i]));
```

Comment:
for example `std::complex<double>`, or sacado type objects, or AdolC, etc.

Reply:
But isn't the additional information lost in that case? Also, a `std::complex` can not be copied to a `double`. So it's not clear to me how this would make it possible to interpolate complex-valued initial conditions.

Reply:
@bangerth The variant for `std::complex<double>` is simply not yet implemented.